deluge/ui/baseargparser.py

```diff
@@ -184,20 +186,47 @@ def __init__(self, *args, **kwargs):
     def parse_args(self, args=None):
         """Parse UI arguments and handle common and process group options.
 
-        Unknown arguments return an error and resulting usage text.
+        Notes:
+            Unknown arguments results in usage text printed and system exit.
+
+        Args:
+            args (list, optional): The arguments to parse.
+
+        Returns:
+            argparse.Namespace: The parsed arguments.
+
         """
         options = super(BaseArgParser, self).parse_args(args=args)
-        return self._parse_args(options)
+        return self._handle_ui_options(options)
 
-    def parse_known_ui_args(self, args=None):
+    def parse_known_ui_args(self, args, withhold=None):
         """Parse UI arguments and handle common and process group options without error.
+
+        Args:
+            args (list): The arguments to parse.
+            withhold (list): Values to ignore in the args list.
+
+        Returns:
+            argparse.Namespace: The parsed arguments.
+
         """
+        if withhold:
+            args = [a for a in args if a not in withhold]
         options, remaining = super(BaseArgParser, self).parse_known_args(args=args)
         options.remaining = remaining
-        return self._parse_args(options)
+        # Hanlde common and process group options
+        return self._handle_ui_options(options)
+
+    def _handle_ui_options(self, options):
+        """Parse UI arguments and handle common and process group options without error.
```

Comment:
Should this be `Handle UI args...`, I don't think a third method with almost identical summary is a good idea :)